chirps/asset/providers/pinecone.py

```diff
@@ -54,12 +55,12 @@ def search(self, query: list, max_results: int) -> list[SearchResult]:
 
         return result_content
 
-    def test_connection(self) -> bool:
+    def test_connection(self) -> PingResult:
         """Ensure that the Pinecone asset can be connected to."""
         try:
             pinecone_lib.init(api_key=self.api_key, environment=self.environment)
-            pinecone_lib.deinit()
-            return True
+            pinecone_lib.list_indexes()
```

Comment:
I think something like this is preferable to make sure that the user-specified index exists
```suggestion
            pinecone_lib.describe_index(self.index_name)
```
you can also except a more specific exception like `pinecone.core.client.exceptions.NotFoundException`